File: Session-3/DDD-Sample-codes/rtgs-return-management-springboot/src/main/java/com/bank/rtgs/returnmanagement/domain/model/ReturnReason.java
package com.bank.rtgs.returnmanagement.domain.model;

import java.util.Objects;

public record ReturnReason(String code, String description) {
    public ReturnReason {
        Objects.requireNonNull(code, "Reason code is required.");
        Objects.requireNonNull(description, "Reason description is required.");
        if (code.isBlank()) throw new IllegalArgumentException("Reason code is required.");
        if (description.isBlank()) throw new IllegalArgumentException("Reason description is required.");
    }
}

File: Session-3/DDD-Sample-codes/rtgs-return-management-dotnet10/Domain/Model/ReturnReason.cs
namespace Rtgs.ReturnManagement.Api.Domain.Model;

public sealed record ReturnReason
{
    public string Code { get; }
    public string Description { get; }

    public ReturnReason(string code, string description)
    {
        if (string.IsNullOrWhiteSpace(code))
            throw new ArgumentException("Reason code is required.");
        if (string.IsNullOrWhiteSpace(description))
            throw new ArgumentException("Reason description is required.");
        Code = code;
        Description = description;
    }
}
